File: packages/core/tests/fixtures/python/code/functions/basic_functions.py
"""
Basic function definitions
Tests: function declarations, parameters, return values
"""

def greet(name: str) -> str:
    return f"Hello, {name}!"


def add(a: int, b: int) -> int:
    return a + b


def process_user(name: str, age: int, active: bool = True) -> dict:
    return {
        "name": name,
        "age": age,
        "active": active,
    }


def multiple_returns(value: int) -> str:
    if value > 0:
        return "positive"
    elif value < 0:
        return "negative"
    else:
        return "zero"


def call_chain():
    data = fetch_data()
    result = transform_data(data)
    return result


def fetch_data():
    return {"value": 42}


def transform_data(data: dict):
    return data["value"] * 2

File: packages/core/tests/fixtures/rust/code/functions/basic_functions.rs
// Basic function definitions
// Tests: function declarations, parameters, return types

fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

fn add(a: i32, b: i32) -> i32 {
    a + b
}

fn multiply(a: i32, b: i32) -> i32 {
    a * b
}

fn is_even(n: i32) -> bool {
    n % 2 == 0
}

fn call_chain() -> i32 {
    let data = fetch_data();
    transform_data(data)
}

fn fetch_data() -> i32 {
    42
}

fn transform_data(value: i32) -> i32 {
    value * 2
}

pub fn main() {
    let result = add(5, 3);
    println!("Result: {}", result);
}
